programs/clearing_house/src/controller/orders.rs

```diff
@@ -951,25 +951,26 @@ pub fn fulfill_order_with_amm(
         None
     };
 
-    let (potentially_risk_increasing, _, quote_asset_amount, quote_asset_amount_surplus, pnl) =
-        controller::position::update_position_with_base_asset_amount(
-            base_asset_amount,
-            order_direction,
-            market,
-            user,
-            position_index,
-            mark_price_before,
-            now,
-            maker_limit_price,
-        )?;
-
-    controller::position::update_unsettled_pnl(&mut user.positions[position_index], market, pnl)?;
-
-    let mut unsettled_pnl = pnl;
-
     let (order_post_only, order_ts, order_direction) =
         get_struct_values!(user.orders[order_index], post_only, ts, direction);
 
+    let (
+        potentially_risk_increasing,
+        _,
+        quote_asset_amount,
+        quote_asset_amount_surplus,
```

Comment:
do we still need to include the surplus since were tracking the market's pnl through position deltas now?

Reply:
its still used for the terminal state accounting: `total_fee_minus_distributions` etc

i think of the the amm.market_position as the local state accounting

Reply:
cant you get the terminal state accounting from the lp position though? feel like we should aim to simplify

Reply:
from the reservation curve's perspective`quote_asset_amount_surplus` is a fee paid at the time of this trade, which is allocated to `amm.total_fee_minus_distributions`. how i think of it is if the amm charged zero fees (spread+exchange) and everyone closed this is how much is left and thus allocated to the amm